<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PasswordGenerator</class>
 <widget class="QMainWindow" name="PasswordGenerator">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>620</width>
    <height>480</height>
   </rect>
  </property>
  <property name="mouseTracking">
   <bool>false</bool>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <widget class="QGroupBox" name="configuration">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="styleSheet">
       <string notr="true">QGroupBox {
	background-color: #E0FFFF;
	border: 1px solid black;
}</string>
      </property>
      <property name="title">
       <string>Configuration</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="3" column="2">
        <widget class="QComboBox" name="complexity">
         <item>
          <property name="text">
           <string>Easy</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Medium</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Hard</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QSpinBox" name="pass_count">
         <property name="maximum">
          <number>1000000</number>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QPlainTextEdit" name="out"/>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label">
         <property name="layoutDirection">
          <enum>Qt::LeftToRight</enum>
         </property>
         <property name="text">
          <string>Password length</string>
         </property>
         <property name="textFormat">
          <enum>Qt::PlainText</enum>
         </property>
         <property name="scaledContents">
          <bool>false</bool>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="wordWrap">
          <bool>false</bool>
         </property>
         <property name="openExternalLinks">
          <bool>false</bool>
         </property>
         <property name="textInteractionFlags">
          <set>Qt::LinksAccessibleByMouse</set>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Passwords count</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item row="4" column="2">
        <widget class="QPushButton" name="generate">
         <property name="styleSheet">
          <string notr="true">QPushButton {
	background-color: #FAEBD7;
}

QPushButton: pressed {
	background-color: #FFFFFF;
}</string>
         </property>
         <property name="text">
          <string>Generate</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="2" column="2">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Complexity</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QSpinBox" name="pass_len">
         <property name="maximum">
          <number>512</number>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QPushButton" name="clear">
         <property name="text">
          <string>Clear</string>
         </property>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QCheckBox" name="pass2file">
         <property name="text">
          <string>Generate in file</string>
         </property>
        </widget>
       </item>
       <item row="5" column="1">
        <widget class="QLineEdit" name="path2dir">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>path/to/dir</string>
         </property>
         <property name="clearButtonEnabled">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QGroupBox" name="status">
      <property name="styleSheet">
       <string notr="true">QGroupBox {
	background-color: #E0FFFF;
	border: 1px solid black;
}</string>
      </property>
      <property name="title">
       <string>Status</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QProgressBar" name="progress">
         <property name="value">
          <number>24</number>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>620</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <action name="actionTelegram">
   <property name="text">
    <string>Telegram</string>
   </property>
  </action>
  <action name="actionTelegram_2">
   <property name="text">
    <string>Telegram</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
